src/methods/truncated_fourier.jl

```diff
@@ -50,68 +50,61 @@ function surrogenerator(x, method::TFTS, rng = Random.default_rng())
     inverse = plan_irfft(forward*x, length(x))
 
     # Pre-compute 𝓕
-    𝓕 = forward*x
+    𝓕 = forward * x
 
     # Polar coordinate representation of the Fourier transform
     rx = abs.(𝓕)
     ϕx = angle.(𝓕)
     n = length(𝓕)
 
-    x_sorted = sort(x)
-
     # These are updated during iteration procedure
     𝓕new = Vector{Complex{Float64}}(undef, length(𝓕))
     𝓕s = Vector{Complex{Float64}}(undef, length(𝓕))
     ϕs = Vector{Complex{Float64}}(undef, length(𝓕))
 
     init = (forward = forward, inverse = inverse,
         rx = rx, ϕx = ϕx, n = n,
-        x_sorted = x_sorted,
         𝓕new = 𝓕new, 𝓕s = 𝓕s, ϕs = ϕs)
 
-    return SurrogateGenerator(method, x, init, rng)
+    return SurrogateGenerator(method, x, similar(x), init, rng)
 end
 
 function (sg::SurrogateGenerator{<:TFTS})()
-    x = sg.x
+    x, s = sg.x, sg.s
     fϵ = sg.method.fϵ
     L = length(x)
 
     init_fields = (:forward, :inverse,
         :rx, :ϕx, :n,
-        :x_sorted,
         :𝓕new, :𝓕s, :ϕs)
 
     forward, inverse,
         rx, ϕx, n,
-        x_sorted,
         𝓕new, 𝓕s, ϕs = getfield.(Ref(sg.init), init_fields)
 
     # Surrogate starts out as a random permutation of x
-    s = x[StatsBase.sample(sg.rng, 1:L, L; replace = false)]
-    𝓕s .= forward*s
+    s .= x[StatsBase.sample(sg.rng, 1:L, L; replace = false)]
```

Comment:
Since `sample` has an inplace version `sample!`  we could pre-allocate a container for the sampled indices.

Reply:
Good catch. This reduced allocations quite a bit, and also made me realize there was another low-hanging allocation optimization right next to it. Thanks!